### ISSUE TITLE
Fedora update handling, `zfs-release` and signing key rotation

### ISSUE THREAD

**Issue by Lalufu** [openzfs/zfs#14344]
TL;DR: I'm happy to work on fixing this if pointed to where the definitions of the `zfs-release` RPMs are

The OpenZFS project has recently rotated the signing key used to sign RPM files for at least Fedora (possibly other RPM based distributions as well). RPMs for Fedora 36 are still signed with the old signing key, while RPMs for Fedora 37 are signed with the new key.
The OpenZFS project distributes the signing keys in a separate RPM (`zfs-release`), which contains the signing key (in `/etc/pki/rpm-gpg/RPM-GPG-KEY-openzfs`) and a repo file for yum/dnf (in `/etc/yum.repos.d/zfs.repo`). The repo file references the key like this:
```
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-openzfs
```
The `zfs-release` file for Fedora 36 ships the old key, while the `zfs-release` file for Fedora 37 ships the new key, both under the same name.
There are two problems with the current approach:
* The `zfs-release` file is not part of the normal zfs repos. This means that it will not be automatically updated for people who have ZFS installed from RPMs, and hence there is no way to automatically distribute new keys to existing installations. Users currently have to manually install a new `zfs-release` file. Most other external repos as well as the distributions themselves update their `-release` files through the enabled repos.
* Currently updating a RPM based zfs system from Fedora 36 to Fedora 37 is broken (see #14201)

The broken update process stems from how `dnf system-upgrade` works on Fedora. The currently installed system must contain all keys needed to validate the files downloaded for the update. In this case, a Fedora 36 system (full updated) must contain all keys to validate the downloaded Fedora 37 files for all enabled repos. In case of a key change between releases, this means two things:
* The "old" system needs a copy of the signing key for the updated system, in addition to the signing key for the "old" system
* The repo definition file needs to be set up in a way that updating/installing RPMs on release `X` and `X+N` might resolve to different keys.

All of this is (relatively) trivial to fix, fortunately.
* Add all signing keys to the `zfs-release` RPM for all supported Fedora versions, under different names (like `RPM-GPG-KEY-openzfs-2013` for the old key and `RPM-GPG-KEY-openzfs-2022` for the new key)
*  Add symlinks to the `zfs-release` file specifying which Fedora release uses which key (`RPM-GPG-KEY-openzfs-fedora-36` -> `RPM-GPG-KEY-openzfs-2013`, `RPM-GPG-KEY-openzfs-fedora-37` -> `RPM-GPG-KEY-openzfs-2022`)
* Reference the signing key in the repo file using the `$releasever` variable (`gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-openzfs-fedora-$releasever`)
* Add the `zfs-release` to the normal ZFS repos

Users who have not modified the ZFS repo definition files will get both the new set of keys, as well as a new repo file which will then work for updating between releases. Users who have modified the zfs repo definition files will still get the new keys but will have to update the repo definition file themselves, but this is already the case today.

**Comment by 1warpengine** [comment]
Can you list a manual solution to the issue (command by command) in order to fix this issue and allow upgrading from Fedora 35/36 to Fedora 37 ? 
I would like to upgrade since me Fedora 35 is EOL and I would like to go directly to 37 but this issue with ZFS is a problem since I use ZFS on my system.
Thanks!

**Comment by 1warpengine** [comment]
Ok solved with this easy manual step to perform before attempt the upgrade to Fedora 37 (if ZFS is installed): https://github.com/openzfs/zfs/issues/14201#issuecomment-1369669663

**Comment by maci0** [comment]
on systems with a prior install this requires the user to `mv zfs.repo.rpmnew zfs.repo` after the latest release has been installed. otherwise the .repo file will reference a keyfile that doesn't exist anymore

**Comment by Lalufu** [comment]
This is true if the user modified the repo file in some way, yes.
Unfortunately there's really no nice way around this.

**Comment by maci0** [comment]
it happened for me on a system where the repo file was not modified at all

**Comment by tonyhutter** [comment]
@Lalufu can we close this issue since the newer zfs-release RPMs do contain the `RPM-GPG-KEY-openzfs-fedora-X` symlinks?

**Comment by Lalufu** [comment]
I think so, yes

**Comment by maci0** [comment]
Okay... after looking at the docs: 
https://openzfs.github.io/openzfs-docs/Getting%20Started/RHEL-based%20distro/index.html#kabi-tracking-kmod
The install procedure would modify the file.
